src/ShoppingCart/Entity/ShoppingCartEntity.php

```diff
@@ -36,70 +36,83 @@ class ShoppingCartEntity implements ShoppingCartEntityInterface
      * @var float
      */
     protected $price;
-    
-	/**
+
+    /**
+     * @var array
+     */
+    protected $product_properties = array();
+
+    /**
      * @return the $id
      */
     public function getId()
     {
         return $this->id;
     }
 
-	/**
+    /**
      * @return the $product
      */
     public function getProduct()
     {
         return $this->product;
     }
 
-	/**
+    /**
      * @return the $qty
      */
     public function getQty()
     {
         return $this->qty;
     }
 
-	/**
+    /**
      * @return the $price
      */
     public function getPrice()
     {
         return $this->price;
     }
 
-	/**
+    /**
      * @param number $id
      */
     public function setId($id)
     {
         $this->id = $id;
     }
 
-	/**
+    /**
      * @param string $product
      */
     public function setProduct($product)
     {
         $this->product = $product;
     }
 
-	/**
+    /**
      * @param number $qty
      */
     public function setQty($qty)
     {
         $this->qty = $qty;
     }
 
-	/**
+    /**
      * @param number $price
      */
     public function setPrice($price)
     {
         $this->price = $price;
     }
 
+    public function setProductProperties($properties)
```

Comment:
Could you please add phpdocs here. That expected parameter is array actually. Also consider it might be good to force to accept array by `public function setProductProperties(array $properties)`